admin/projects.go

```diff
@@ -218,11 +225,25 @@ func (s *Service) UpdateProject(ctx context.Context, proj *database.Project, opt
 		}
 	}
 
+	projNameUpdated := proj.Name != opts.Name
+
 	proj, err := s.DB.UpdateProject(ctx, proj.ID, opts)
 	if err != nil {
 		return nil, err
 	}
 
+	if !requireNewDeployments && !impactsDeployments && projNameUpdated {
+		org, err := s.DB.FindOrganization(ctx, proj.OrganizationID)
+		if err != nil {
+			return nil, err
+		}
+
+		err = s.UpdateDeploymentAnnotations(ctx, org, proj)
+		if err != nil {
+			return nil, err
+		}
+	}
```

Comment:
1. `requireNewDeployments` can never be true (it returns early if that's the case)
2. What about the case where a project's branch and name is updated in the same request? I'm not sure `impactsDeployments` and `projNameUpdated` are mutually exclusive

Reply:
I introduced annotations into `EditInstanceRequest` so that annotations are updated (if needed) if `impactsDeployments == true`
`UpdateDeploymentAnnotations` is called only if annotations changed and `impactsDeployments == false`